vivarium/framework/lookup.py

```diff
@@ -29,11 +29,31 @@ class InterpolatedTableView(TableView):
     These cannot be created directly. Use the `lookup` method on the builder during setup.
     """
 
-    def __init__(self, interpolation, population_view, clock=None):
-        self.interpolation = interpolation
+    def __init__(self, data, population_view, key_columns, parameter_columns, interpolation_order, clock=None):
+        assert data is not None, "Must supply some data"
+
+        self._data = data
+        self._interpolation = None
         self.population_view = population_view
+        self._key_columns = key_columns
+        self._parameter_columns = parameter_columns
+        self._interpolation_order = interpolation_order
         self.clock = clock
 
+    @property
+    def interpolation(self):
+        if self._interpolation is None:
+            data = self._data
+            if callable(data) and not isinstance(data, Interpolation):
+                data = data()
+
+            if isinstance(data, Interpolation):
```

Comment:
We should probably try to cleanup any places where people are manually creating interpolations.  This feels incorrect to me.